e2e_test.go

```diff
@@ -0,0 +1,1035 @@
+		}
+
+		results, err := client.MultiInsert(ctx, req)
+		if err != nil {
+			t.Fatalf("error: %s", err)
+		}
+
+		if results == nil {
+			t.Fatal("error: returned results is nil")
+		}
+
+		for _, result := range results.GetLocations() {
+			if len(result.GetIps()) != 1 {
+				t.Fatal("error: invalid number of ips")
+			}
+		}
+	})
+
+	t.Run(`Test for StreamInsert operation`, func(t *testing.T) {
+		beforeEach()
+		defer afterEach()
+
+		stream, err := client.StreamInsert(ctx)
+		if err != nil {
+			t.Fatalf("error: %s", err)
+		}
+
+		wg := new(sync.WaitGroup)
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			for {
+				result, err := stream.Recv()
+				if err != nil {
+					if err == io.EOF {
+						return
+					}
+
+					t.Errorf("error: %s", err)
+
+					continue
+				}
+
+				if result == nil {
+					t.Error("error: returned results is nil")
+					continue
+				}
+
+				loc := result.GetLocation()
+				if loc == nil {
+					if result.GetStatus().GetCode() != 0 {
+						t.Errorf("result code is %d", result.GetStatus().GetCode())
+						continue
+					}
+
+					t.Error("location is nil")
+					continue
+				}
+
+				if len(loc.GetIps()) != 1 {
+					t.Error("error: invalid number of ips")
+				}
+			}
+		}()
+
+		for i := 11; i < 100; i++ {
+			req := &payload.Insert_Request{
+				Vector: &payload.Object_Vector{
+					Id:     data[i].ID,
+					Vector: data[i].Vector,
+				},
+				Config: insertConfig,
+			}
+
+			err = stream.Send(req)
+			if err != nil {
+				t.Errorf("error: %s", err)
+			}
+		}
+
+		stream.CloseSend()
+
+		wg.Wait()
+	})
+
+	// Agent
+
+	t.Run(`Test for CreateIndex operation`, func(t *testing.T) {
+		beforeEach()
+		defer afterEach()
+
+		req := &payload.Control_CreateIndexRequest{
+			PoolSize: 10000,
+		}
+
+		_, err := agentClient.CreateIndex(ctx, req)
+		if err != nil {
+			t.Errorf("error: %s", err)
+		}
+	})
+
+	t.Run(`Test for SaveIndex operation`, func(t *testing.T) {
+		beforeEach()
+		defer afterEach()
+
+		req := &payload.Empty{}
+
+		_, err := agentClient.SaveIndex(ctx, req)
+		if err != nil {
+			t.Errorf("error: %s", err)
+		}
+	})
+
+	t.Run(`Test for IndexInfo operation`, func(t *testing.T) {
+		beforeEach()
+		defer afterEach()
+
+		req := &payload.Empty{}
+
+		result, err := agentClient.IndexInfo(ctx, req)
+		if err != nil {
+			t.Errorf("error: %s", err)
+		}
+
+		if result.GetStored() != 100 {
+			t.Errorf("stored is not same as expected. received: %d, expected: %d", result.GetStored(), 100)
+		}
+
+		if result.GetUncommitted() != 0 {
+			t.Errorf("uncommitted is not same as expected. received: %d, expected: %d", result.GetUncommitted(), 0)
+		}
+	})
+
+	// Exists
+
+	t.Run(`Test for Exists operation`, func(t *testing.T) {
+		beforeEach()
+		defer afterEach()
+
+		req := &payload.Object_ID{
+			Id: data[0].ID,
+		}
+
+		result, err := client.Exists(ctx, req)
+		if err != nil {
+			t.Errorf("error: %s", err)
+		}
+
+		if result.GetId() == "" {
+			t.Error("expected ID is not found.")
```

Comment:
Should we remove the last period of last to unify with others? 🤔 

```suggestion
			t.Error("expected ID is not found")
```